packages/hdwallet-ledger/src/ethereum.ts

```diff
@@ -1,45 +1,84 @@
+import { ecrecover, hashPersonalMessage, toBuffer, pubToAddress, sha256, ripemd160 } from 'ethereumjs-util'
+import * as core from '@shapeshiftoss/hdwallet-core'
+import { LedgerTransport } from './transport'
 import {
-  ecrecover,
-  hashPersonalMessage,
-  toBuffer,
-  pubToAddress
-} from 'ethereumjs-util'
+  createXpub,
+  compressPublicKey,
+  encodeBase58Check,
+  networksUtil,
+  parseHexString,
+  handleError
+} from './utils'
+
 // @ts-ignore
 import * as Ethereumjs from 'ethereumjs-tx'
 const { default: EthereumTx } = Ethereumjs as any
-import {
-  addressNListToBIP32,
-  Constructor,
-  ETHWallet,
-  ETHGetAddress,
-  ETHSignTx,
-  ETHSignedTx,
-  ETHGetAccountPath,
-  ETHAccountPath,
-  ETHSignMessage,
-  ETHSignedMessage,
-  ETHVerifyMessage,
-  toHexString,
-  stripHexPrefixAndLower,
-  slip44ByCoin
-} from '@shapeshiftoss/hdwallet-core'
-import { handleError } from './utils'
-import { LedgerTransport } from './transport'
 
 export async function ethSupportsNetwork(chain_id: number): Promise<boolean> {
   return chain_id === 1
 }
 
-export async function ethGetAddress(transport: LedgerTransport, msg: ETHGetAddress): Promise<string> {
-  const bip32path = addressNListToBIP32(msg.addressNList)
+export async function ethGetAddress(transport: LedgerTransport, msg: core.ETHGetAddress): Promise<string> {
+  const bip32path = core.addressNListToBIP32(msg.addressNList)
   const res = await transport.call('Eth', 'getAddress', bip32path, !!msg.showDisplay)
   handleError(res, transport, 'Unable to obtain ETH address from device.')
 
   return res.payload.address
 }
 
-export async function ethSignTx(transport: LedgerTransport, msg: ETHSignTx): Promise<ETHSignedTx> {
-  const bip32path = addressNListToBIP32(msg.addressNList)
+// Adapted from https://github.com/LedgerHQ/ledger-wallet-webtool
+export async function ethGetPublicKeys (transport: LedgerTransport, msg: Array<core.GetPublicKey>): Promise<Array<core.PublicKey>> {
+    const xpubs = []
+
+    for (const getPublicKey of msg) {
+      let { addressNList, coin, symbol, scriptType } = getPublicKey
+
+      if (symbol !== 'ETH') {
```

Comment:
I think this test should match what their device actually cares about, which I think is that `addressNList` must start with  `[0x80000000 + 44, 0x80000000 + 60]` if you're using the Eth app. Then we can avoid adding `symbol` to `GetPublicKey`, which seems odd (not sure why it's being passed in that other spot, but it shouldn't be what we're using here).